department-of-veterans-affairs/va.gov-cms
# VACMS-14405 fix jsonapi responses

## Description

Relates to #14405. This PR relates to next-build functionality -- the CMS's JSONAPI endpoint wasn't returning all the expected data without these two patches included. 

I also needed to remove the config for news story node's path value... something apparently related to base field override (possibly translations?) seems to mysteriously drop some fields in the json response. The BaseFieldOverride patch fixes that behavior, but the config was committed prior. Removing that config shows the path variable in the response as expected. Other content types may need their `core.base_field_override.node.NODE_TYPE.path.yml` files removed in the future for JSONAPI as well.


## Testing done
running locally, hitting a jsonapi endpoint for a news story node should include a `path` object underneath the `attributes` key:
https://va-gov-cms.ddev.site/jsonapi/node/news_story/1dbd5b1e-0fb2-43fb-8581-a2f9edfe3c83

## Screenshots
![Screenshot 2023-07-13 at 1 36 34 PM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/11279744/c81e4f91-afbb-4a50-9793-23ed00380187)


## QA steps
- What needs to be checked to prove this works?  
Visit the same endpoint without the patches from this PR applied and see that the path variable is missing.

You could also try running the next-build repo as-is. Story listing pages will fail because `entity.path.alias` is not found in the data for any of the listed news story nodes. There are other cleanup tasks pushed to that repo here: https://github.com/department-of-veterans-affairs/next-build/pull/126

- What needs to be checked to prove it didn't break any related things?  
- [ ] Do News Stories still translate as expected?
- [ ] Validate that the path of a newly created news story matches that of a newly created news story on staging.

### Definition of Done


### Select Team for PR review



### CMS user-facing announcement
Is an announcement needed to let editors know of this change?
